pkg/virt-handler/device-manager/device_plugin_base.go

```diff
@@ -68,7 +71,9 @@ func (dpi *DevicePluginBase) ListAndWatch(_ *pluginapi.Empty, s pluginapi.Device
 					dev.Health = devHealth.Health
 				}
 			}
-			s.Send(&pluginapi.ListAndWatchResponse{Devices: dpi.devs})
+			if err := s.Send(&pluginapi.ListAndWatchResponse{Devices: dpi.devs}); err != nil {
+				return err
```

Comment:
I think you don't want to exit here in the middle of the endless loop on an error. I think it might be better to instead log the error and either re-try sending or simply continue.

Reply:
It wasn't exit in the previous version, so, yes we can keep it that way